GVRf/Framework/jni/objects/mesh.cpp

```diff
@@ -204,6 +205,46 @@ void Mesh::getTransformedBoundingBoxInfo(glm::mat4 *Mat,
     }
 }
 
+// This gives us a really coarse bounding sphere given the already calcuated bounding box.  This won't be a tight-fitting sphere because it is based on the bounding box.  We can revisit this later if we decide we need a tighter sphere.
+const float *Mesh::getBoundingSphereInfo() {
+    if (have_bounding_box_ == false) {
+        getBoundingBoxInfo();
+    }
+
+    if (have_bounding_sphere_ == true) {
+        return bounding_sphere_info_;
+    }
+
+    // get the bounding box into nicely readable variables
+    float min_x = bounding_box_info_[0];
+    float max_x = bounding_box_info_[3];
+    float min_y = bounding_box_info_[1];
+    float max_y = bounding_box_info_[4];
+    float min_z = bounding_box_info_[2];
+    float max_z = bounding_box_info_[5];
+
+    // find center
+    float center_x = (min_x + max_x) / 2.0f;
+    float center_y = (min_y + max_y) / 2.0f;
+    float center_z = (min_z + max_z) / 2.0f;
+
+    // find radius
+    float x_squared = (min_x - center_x) * (min_x - center_x);
+    float y_squared = (min_y - center_y) * (min_y - center_y);
+    float z_squared = (min_z - center_z) * (min_z - center_z);
+    float radius = sqrtf(x_squared + y_squared + z_squared);
+
+    // assign the sphere
+    bounding_sphere_info_[0] = center_x;
+    bounding_sphere_info_[1] = center_y;
+    bounding_sphere_info_[2] = center_z;
+    bounding_sphere_info_[3] = radius;
+
+    have_bounding_sphere_ = true;
+
+    return bounding_sphere_info_;
+}
```

Comment:
It is not that related to this pull request but I am somehow thinking in deleteVaos, you may need to reset the have_bounding_box and have_bounding_sphere since all data are considered dirty with the dirty flag Mihail brought in in his pull request.